src/modules/csm/csm.py

```diff
@@ -0,0 +1,346 @@
+import logging
```

Comment:
This is not a strict requirement but recommendation for the future refactoring
IMHO, the code in CSM module a little bit difficult to read because of the function complexity.

I do not insist on using of any specific automation tool for code complexity measuring but as an example here is report from `radon` tool. Ideally it would be great to simplify cyclomatic complexity of those functions.

![image](https://github.com/user-attachments/assets/2214ba65-e832-4fb5-97ae-001e54750c64)